src/main/internals/buffer-crc32.ts

```diff
@@ -0,0 +1,115 @@
+  0x5edef90e, 0x29d9c998, 0xb0d09822, 0xc7d7a8b4, 0x59b33d17,
+  0x2eb40d81, 0xb7bd5c3b, 0xc0ba6cad, 0xedb88320, 0x9abfb3b6,
+  0x03b6e20c, 0x74b1d29a, 0xead54739, 0x9dd277af, 0x04db2615,
+  0x73dc1683, 0xe3630b12, 0x94643b84, 0x0d6d6a3e, 0x7a6a5aa8,
+  0xe40ecf0b, 0x9309ff9d, 0x0a00ae27, 0x7d079eb1, 0xf00f9344,
+  0x8708a3d2, 0x1e01f268, 0x6906c2fe, 0xf762575d, 0x806567cb,
+  0x196c3671, 0x6e6b06e7, 0xfed41b76, 0x89d32be0, 0x10da7a5a,
+  0x67dd4acc, 0xf9b9df6f, 0x8ebeeff9, 0x17b7be43, 0x60b08ed5,
+  0xd6d6a3e8, 0xa1d1937e, 0x38d8c2c4, 0x4fdff252, 0xd1bb67f1,
+  0xa6bc5767, 0x3fb506dd, 0x48b2364b, 0xd80d2bda, 0xaf0a1b4c,
+  0x36034af6, 0x41047a60, 0xdf60efc3, 0xa867df55, 0x316e8eef,
+  0x4669be79, 0xcb61b38c, 0xbc66831a, 0x256fd2a0, 0x5268e236,
+  0xcc0c7795, 0xbb0b4703, 0x220216b9, 0x5505262f, 0xc5ba3bbe,
+  0xb2bd0b28, 0x2bb45a92, 0x5cb36a04, 0xc2d7ffa7, 0xb5d0cf31,
+  0x2cd99e8b, 0x5bdeae1d, 0x9b64c2b0, 0xec63f226, 0x756aa39c,
+  0x026d930a, 0x9c0906a9, 0xeb0e363f, 0x72076785, 0x05005713,
+  0x95bf4a82, 0xe2b87a14, 0x7bb12bae, 0x0cb61b38, 0x92d28e9b,
+  0xe5d5be0d, 0x7cdcefb7, 0x0bdbdf21, 0x86d3d2d4, 0xf1d4e242,
+  0x68ddb3f8, 0x1fda836e, 0x81be16cd, 0xf6b9265b, 0x6fb077e1,
+  0x18b74777, 0x88085ae6, 0xff0f6a70, 0x66063bca, 0x11010b5c,
+  0x8f659eff, 0xf862ae69, 0x616bffd3, 0x166ccf45, 0xa00ae278,
+  0xd70dd2ee, 0x4e048354, 0x3903b3c2, 0xa7672661, 0xd06016f7,
+  0x4969474d, 0x3e6e77db, 0xaed16a4a, 0xd9d65adc, 0x40df0b66,
+  0x37d83bf0, 0xa9bcae53, 0xdebb9ec5, 0x47b2cf7f, 0x30b5ffe9,
+  0xbdbdf21c, 0xcabac28a, 0x53b39330, 0x24b4a3a6, 0xbad03605,
+  0xcdd70693, 0x54de5729, 0x23d967bf, 0xb3667a2e, 0xc4614ab8,
+  0x5d681b02, 0x2a6f2b94, 0xb40bbe37, 0xc30c8ea1, 0x5a05df1b,
+  0x2d02ef8d
+]
+
+if (typeof Int32Array !== 'undefined') {
+  CRC_TABLE = new Int32Array(CRC_TABLE)
+}
+
+function newEmptyBuffer(length: number) {
+  const buffer = Buffer.alloc(length)
+  buffer.fill(0x00)
+  return buffer
+}
+
+function ensureBuffer(input: Buffer | string | number) {
+  if (Buffer.isBuffer(input)) {
+    return input
+  }
+
+  const hasNewBufferAPI =
+      typeof Buffer.alloc === 'function' &&
+      typeof Buffer.from === 'function'
+
+  if (typeof input === 'number') {
+    return hasNewBufferAPI ? Buffer.alloc(input) : newEmptyBuffer(input)
+  } else if (typeof input === 'string') {
+    return hasNewBufferAPI ? Buffer.from(input) : new Buffer(input)
+  } else {
+    throw new Error('input must be buffer, number, or string, received ' +
+                    typeof input)
+  }
+}
+
+function bufferizeInt(num: number) {
+  const tmp = ensureBuffer(4)
+  tmp.writeInt32BE(num, 0)
+  return tmp
+}
+
+function _crc32(buf: string | Buffer, previous?: Buffer | number): number {
+  buf = ensureBuffer(buf)
+  if (Buffer.isBuffer(previous)) {
+    previous = previous.readUInt32BE(0)
+  }
+  let crc = ~~previous ^ -1
+  for (let n = 0; n < buf.length; n++) {
+    crc = CRC_TABLE[(crc ^ buf[n]) & 0xff] ^ (crc >>> 8)
+  }
+  return crc ^ -1
+}
+
+function crc32() {
```

Comment:
Other than vendoring some libraries that are not used often, I wanted to vendor some libraries that not really maintained anymore and might contain vulnerabilities.

This crc32 library is simple enough that I decided just to do that. And yes, we should inlude a unit test for the specific libraries that we vendored.